sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs

```diff
@@ -210,8 +220,14 @@ private MutableJsonDocument(JsonDocument document, ReadOnlyMemory<byte> utf8Json
             _serializerOptions = serializerOptions ?? DefaultSerializerOptions;
         }
 
+#if !NET5_0
```

Comment:
Do we need the ifdef here - `UnconditionalSuppressMessage` should be allowed everywhere even in .NET 5.